### ISSUE TITLE
Better Download Error Response

### ISSUE THREAD

**Issue by littlebtc** [littlebtc/nicofox#3]
[Old Google Code Issue about these.](http://code.google.com/p/nicofox/issues/detail?id=34) (Chinese, sorry.)

There are lots of errors can be happened in the download process (mainly, in  `DownloadUtils.jsm` and `VideoInfoReader.jsm`), and lots of them will stop all download items and make a modal dialog. It should be fixed:
- Videos is removed/unavailable/under country restriction: Some is now filtered with simple video info (by calling the XML API), but in some cases, we will know whether the video is available only after reading the video page in the (non-simple) video info reader. In this case, mark the video download as failed, and continue to download the next item in the queue.
- Anti-flood is detected in the (non-simple) video info reader: Pause the download queue for 60~90 secs (If pause in #1 is implemented), or cancel all downloads (If not implemented). A complex routine may be required to do this :(
- Video file downloads are incomplete: Cleanup file (implemented), mark the video download as failed, and continue to download the next item in the queue.

By the way there should be a better download error notification. But it is ok to separate it to another issue.

This work is assigned.
